src/transforms/code-scanning-alert.ts

```diff
@@ -0,0 +1,100 @@
+import issueKeyParser from "jira-issue-key-parser";
+import {Context} from "probot/lib/context";
+import {JiraRemoteLinkData, JiraRemoteLinkStatusAppearance} from "../interfaces/jira";
+import logger from "../config/logger";
+
+const MAX_STRING_LENGTH = 255;
+
+const getPullRequestTitle = (repoName: string, prId: number, repoOwner: string, context: Context): Promise<string> => {
+	return context.github.pulls.get({
+		owner: repoOwner,
+		repo: repoName,
+		pull_number: prId
+	}).then((response) => {
+		if (response.status !== 200) {
+			throw response.data;
+		} else {
+			return response.data.title;
+		}
+	}).catch((err) => {
+		logger.warn(`Received error when querying for Pull Request information: ${JSON.stringify(err)}`);
+		return Promise.resolve("");
+	})
+}
+
+const getEntityTitle = (ref: string, repoName: string, repoOwner: string, context: Context): Promise<string> => {
+	// ref can either be a branch reference or a PR reference
+	const components = ref.split("/")
+	switch (components[1]) {
+		case "heads": // branch
+			// The branch name may contain forward slashes! Rejoin them
+			return Promise.resolve(components.slice(2).join("/"));
```

Comment:
If branch name is: "issue/abc-124/my-awesome-branch" we'll get "my-awesome-branch" 

I wonder is it what we want?

Reply:
We are not splitting slashes on the branch name, but instead on the `ref` string. This `ref` string will look something like:

`refs/heads/<branchname>`

Let's say the branch name is `issue/abc-124/my-awesome-branch`. Then components will be `[refs, heads, issue, abc-124, my-awesome-branch]`. We split up by slash so that we can confirm that the ref is referencing a branch head. But you are right that branches can have slashes, so we need to join back the branch name by joining `components.slice(2)` (grab everything after `heads`)